elfi/examples/scratch_assay.py

```diff
@@ -0,0 +1,192 @@
+    num_obs = int(num_iter/obs_interval)
+    obs_arr = np.zeros((num_obs+1, nrows, ncols))
+    obs_arr[0] = np.copy(cell_arr)
+
+    for iteration in range(num_iter):
+
+        # update cell count and coordinates
+
+        num_cells = int(np.sum(cell_arr))
+        cell_coords = np.transpose(np.array(np.where(cell_arr)))
+
+        if num_cells == nrows * ncols:
+            continue
+
+        # motility events
+
+        candidates = random_state.choice(num_cells, size=num_cells)
+        p = random_state.uniform(size=num_cells)
+        candidates = candidates[p < pm]
+
+        for cell in candidates:
+            coords = _random_move(cell_coords[cell], nrows, ncols, random_state)
+            if cell_arr[coords[0], coords[1]] == 0:
+                # move
+                cell_arr[cell_coords[cell][0], cell_coords[cell][1]] = 0
+                cell_arr[coords[0], coords[1]] = 1
+                # update coordinate list
+                cell_coords[cell] = coords
+
+        # proliferation events
+
+        candidates = random_state.choice(num_cells, size=num_cells)
+        p = random_state.uniform(size=num_cells)
+        candidates = candidates[p < pp]
+
+        for cell in candidates:
+            coords = _random_move(cell_coords[cell], nrows, ncols, random_state)
+            cell_arr[coords[0], coords[1]] = 1   # increases cell count if coords is unoccupied
+
+        # observations
+
+        if (iteration + 1) % obs_interval == 0:
+            obs_arr[int((iteration + 1)/obs_interval)] = np.copy(cell_arr)
+
+    return np.transpose(obs_arr, (1, 2, 0))
+
+
+def cell_summaries(x):
+    """Calculate summary statistics proposed by Price et al (2018).
+
+    Parameters
+    ----------
+    x : np.ndarray
+        Simulated/observed data in shape (batch_size, nrows, ncols, num_obs).
+
+    Returns
+    -------
+    np.ndarray (batch_size, num_obs+1)
+
+    """
+    # 1. mismatch between consecutive observations
+    ds = np.sum(np.abs((x[:, :, :, : -1] - x[:, :, :, 1:])), axis=(1, 2))
+
+    # 2. final cell count
+    count = np.sum(x[:, :, :, -1], axis=(1, 2))[:, None]
+
+    return np.concatenate((ds, count), axis=1)
+
+
+def get_model(true_params=None, init_arr=None, init_params=None, seed_obs=None):
+    """
+    Return complete scratch assay model.
+
+    Parameters
+    ----------
+    true_params : list, optional
+        Parameters with which the observed data is generated.
+    init_arr : np.ndarray, optional
+        Initial cell locations as binary array.
+    init_params : list, optional
+        Parameters for random initialisation. Not used if init_arr is provided.
+    seed_obs : int, optional
+        Seed for the observed data generation.
+
+    Returns
+    -------
+    elfi.ElfiModel
+
+    """
+    if true_params is None:
+        true_params = [0.25, 0.002]
+
+    m = elfi.ElfiModel()
+
+    # priors
+    elfi.Prior('uniform', 0, 1, model=m, name='pm')
+    elfi.Prior('uniform', 0, 0.01, model=m, name='pp')
```

Comment:
Both given references use [0,1] priors. Would it make sense use them, or does it lead to computational issues?

Reply:
true. there should be no computational issues, but i made a mistake and now the cell array is not observed when it becomes full. i will fix that and we can also update the prior. the narrower prior that was used here is from [thomas et al](https://doi.org/10.1214/20-BA1238).